client/src/index.tsx

```diff
@@ -10,6 +10,87 @@ import { GleanProvider } from "./telemetry/glean-context";
 
 // import * as serviceWorker from './serviceWorker';
 
+let attemptedFallback = false;
```

Comment:
Could we put this in a function, extract it to a separate file, and call that function here?

The function should have 1-3 comment lines before to explain why it's necessary.